data-contracts/README.md

```diff
@@ -0,0 +1,439 @@
+```
+
+Now implement classes to produce and consume events with this schema. First the producer class, extending `BaseProducer`:
+
+```kotlin
+class MembershipProducer: BaseProducer<String, Membership>(mapOf(
+    ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG to "org.apache.kafka.common.serialization.StringSerializer",
+    ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG to "io.confluent.kafka.serializers.KafkaAvroSerializer",
+    ProducerConfig.CLIENT_ID_CONFIG to "membership-producer-app-v1",
+    AbstractKafkaSchemaSerDeConfig.LATEST_COMPATIBILITY_STRICT to true,
+    AbstractKafkaSchemaSerDeConfig.USE_LATEST_VERSION to false,
+    AbstractKafkaSchemaSerDeConfig.USE_LATEST_WITH_METADATA to "major_version=1"
+))
+```
+
+This implementation must pass the configuration parameters required to serialize the key and value of events to the `membership-avro` topic. In this use case, the key is a `String` and the value is of type `Membership` - generated from the schema. The value uses the schema-registry-aware `Serializer` implementation `KafkaAvroSerializer`. The underlying serializer is also configured to NOT use the latest version of the schema (`use.latest.version`) and to relax the strict compatibility checks (`latest.compatibility.strict`). For more on these settings, see the CC doc's explanation of [the differences between preregistered and client-derived schemas](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/index.html#handling-differences-between-preregistered-and-client-derived-schemas).
+
+Let's also implement the `BaseConsumer` class, configured to use this `major_version` of the schema:
+
+```kotlin
+class MembershipConsumer: BaseConsumer<String, Membership>(mapOf(
+    ConsumerConfig.GROUP_ID_CONFIG to "app-schema-v1",
+    ConsumerConfig.AUTO_OFFSET_RESET_CONFIG to "earliest",
+    ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG to "org.apache.kafka.common.serialization.StringDeserializer",
+    ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG to "io.confluent.kafka.serializers.KafkaAvroDeserializer",
+    AbstractKafkaSchemaSerDeConfig.LATEST_COMPATIBILITY_STRICT to true,
+    AbstractKafkaSchemaSerDeConfig.USE_LATEST_VERSION to false,
+    AbstractKafkaSchemaSerDeConfig.USE_LATEST_WITH_METADATA to "major_version=1"
+)) {
+
+    override fun consumeRecord(
+        key: String,
+        value: Membership
+    ) {
+        logger.info("Received Membership ${key}, ${value}")
+    }
+}
+```
+
+The `consumeRecord` function is where we would typically start the business logic of actually "consuming" Kafka events. This implementation simply logs the consumed records to the provided `Logger` instance in the superclass.
+
+To exercise these producer and consumer implementations, we created a `main` function in the `ApplicationMain` class to start a consumer instance and a producer instance periodically send random events to the `membership-avro` topic.
+
+<details>
```

Comment:
i'd remove this. just reference the code with a link and say what it does in English, but no need to put code inline.